source/dofs/dof_accessor_set.cc

```diff
@@ -41,91 +41,235 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+template <typename Number>
+DeclException2(ExcNonMatchingElementsSetDofValuesByInterpolation,
+               Number,
+               Number,
+               << "Called set_dof_values_by_interpolation(), but"
+               << " the element to be set, value " << std::setprecision(16)
+               << arg1 << ", does not match with the value "
+               << std::setprecision(16) << arg2 << " already set before.");
+
+namespace internal
```

Comment:
Although the following functions are all internal, would you go the extra mile to write a short paragraph for each of them for documentation?

For example, I get why you have two `get_abs` functions, but maybe you can briefly write for which case these functions are designed (also, why the use of `NumberTraits` is only necessary for the first case).

I know this means extra work, but I'm sure it will make maintenance in the future a lot easier.